lib/socket.io/router.js

```diff
@@ -52,18 +52,25 @@ Router.prototype.middleware = function () {
     socket._onevent = socket.onevent;
     socket.onevent = function (packet) {
       var args = packet.data || [];
+
       if (!args.length) {
         debug('event args not exist');
         return socket._onevent(packet);
       }
 
+      if(null != packet.id) {
+        debug('attaching ack callback to event')
+        args.push(socket.ack(packet.id))
+      }
+      
       context.event = args[0];
       context.data = args.slice(1);
 
-      co.wrap(gen).call(context, function (err) {
-        if (err) console.error(err.stack);
-        //TODO: error
```

Comment:
@ariporad I didn't see any specific way of dealing with errors within the library, I assumed I should not force any specific way before the library was further developed, as I don't know how you were intending to manage this. Actually, my bad for not keeping your comment there - it was meant to be transferred to the .catch() callback further down.